src/Symfony/Bridge/PhpUnit/SymfonyTestsListener.php

```diff
@@ -197,6 +197,11 @@ public function endTest(\PHPUnit_Framework_Test $test, $time)
                 DnsMock::withMockedHosts(array());
             }
         }
+
+        if ($test instanceof \PHPUnit_Framework_TestCase && 0 === strpos($test->getName(), 'testLegacy') && empty($test->getTestResultObject()->warningCount())) {
```

Comment:
when the annotation is in place, we should not trigger the warning. What about the other cases?
From DeprecationErrorHandler:
```php
                } elseif (0 === strpos($method, 'testLegacy')
                    || 0 === strpos($method, 'provideLegacy')
                    || 0 === strpos($method, 'getLegacy')
                    || strpos($class, '\Legacy')
```